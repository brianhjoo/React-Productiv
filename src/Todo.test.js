import React from "react";
import { render } from "@testing-library/react";
import Todo from './Todo';


const props = {
  title: 'Finish project',
  priority: '1',
  description: 'Complete all remaining tasks',
};


describe ("productiv app", function() {
  it('renders without crashing', function() {
    render(<Todo />);
  });

  it('displays the correct title', function() {
    const result = render(<Todo title={props.title} />);
    expect(result.queryByText('Finish project')).toBeInTheDocument();
  });

  it('displays the correct description', function() {
    const result = render(<Todo description={props.description} />);
    expect(result.queryByText('Complete all remaining tasks')).toBeInTheDocument();
  });

  it('displays correct priority and data type', function() {
    const result = render(<Todo priority={props.priority} />);
    expect(result.queryByText('1')).not.toBeInTheDocument();
  });

  it('matches snapshot', function() {
    const { container } = render(<Todo />);
    expect(container).toMatchSnapshot();
  });
});