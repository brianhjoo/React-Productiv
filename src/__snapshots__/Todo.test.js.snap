// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b />
       
      <small>
        (priority: 
        )
      </small>
    </div>
    <div>
      <small />
    </div>
  </div>
</div>
`;
